cirq-core/cirq/ops/pauli_string_phasor.py

```diff
@@ -58,26 +60,62 @@ def __init__(
                 in the form of the t in (-1)**t = exp(i pi t).
 
         Raises:
-            ValueError: If the given pauli string does not have eignevalues +1
-                or -1.
+            ValueError: If coefficient is not 1 or -1.
         """
-        if pauli_string.coefficient == -1:
-            pauli_string = -pauli_string
-            exponent_pos, exponent_neg = exponent_neg, exponent_pos
+        gate = PauliStringPhasorGate(
+            dps.DensePauliString(pauli_string.values(), coefficient=pauli_string.coefficient),
+            exponent_neg=exponent_neg,
+            exponent_pos=exponent_pos,
+        )
+        super().__init__(gate, pauli_string.qubits)
+        self._pauli_gate = gate
+        self._pauli_string = gate.dense_pauli_string.on(*self.qubits)
 
-        if pauli_string.coefficient != 1:
-            raise ValueError(
-                "Given PauliString doesn't have +1 and -1 eigenvalues. "
-                "pauli_string.coefficient must be 1 or -1."
-            )
+    @property
+    def exponent_neg(self):
+        return self._pauli_gate.exponent_neg
+
+    @exponent_neg.setter  # type: ignore
+    @deprecated(
+        deadline="v0.15",
+        fix="The mutators of this class are deprecated, instantiate a new object instead.",
+    )
+    def exponent_neg(self, exponent_neg):
+        # coverage: ignore
+        self._pauli_gate._exponent_neg = exponent_neg
+
+    @property
+    def exponent_pos(self):
+        return self._pauli_gate.exponent_pos
+
+    @exponent_pos.setter  # type: ignore
+    @deprecated(
+        deadline="v0.15",
+        fix="The mutators of this class are deprecated, instantiate a new object instead.",
+    )
+    def exponent_pos(self, exponent_pos):
+        # coverage: ignore
+        self._pauli_gate._exponent_pos = exponent_pos
 
-        super().__init__(pauli_string)
-        self.exponent_neg = value.canonicalize_half_turns(exponent_neg)
-        self.exponent_pos = value.canonicalize_half_turns(exponent_pos)
+    @property
+    def pauli_string(self):
+        return self._pauli_string
+
+    @pauli_string.setter  # type: ignore
+    @deprecated(
+        deadline="v0.15",
+        fix="The mutators of this class are deprecated, instantiate a new object instead.",
+    )
+    def pauli_string(self, pauli_string):
```

Comment:
Ideally, `self.qubits` can also potentially change when updating the underlying pauli string. Big +1 on deprecating this but.